src/app/view/Lock/useLockScreenWrapper.ts

```diff
@@ -0,0 +1,67 @@
+import { EventEmitter } from 'events'
+
+import { useEffect, useState } from 'react'
+
+const lockScreenEventHandler = new EventEmitter()
+
+const lockScreenEvents = {
+  SHOW: 'SHOW',
+  HIDE: 'HIDE'
+} as const
+
+export const lockScreens = {
+  LOCK_SCREEN: 'LOCK_SCREEN',
+  PASSWORD_PROMPT: 'PASSWORD_PROMPT',
+  PIN_PROMPT: 'PIN_PROMPT',
+  SET_PASSWORD: 'SET_PASSWORD',
+  SET_PIN: 'SET_PIN'
+} as const
+
+export type LockScreenEnum = keyof typeof lockScreens
+
+export const showSecurityScreen = (screen: LockScreenEnum): void => {
+  lockScreenEventHandler.emit(lockScreenEvents.SHOW, screen)
+}
+export const hideSecurityScreen = (screen: LockScreenEnum): void => {
+  lockScreenEventHandler.emit(lockScreenEvents.HIDE, screen)
+}
+
+interface LockScreenState {
+  currentSecurityScreen: LockScreenEnum | undefined
+}
+
+export const useLockScreenWrapper = (): LockScreenState => {
+  const [currentScreen, setCurrentScreen] = useState<
```

Comment:
why not directly use a reducer hook here (instead of useState)? seeing the logic in the component is strongly reminiscent of a reducer structure

Reply:
What are the advantages of using a reducer here? My thoughts was that reducers were only useful when having a complex object in the state. Here we have only a string in it.

Reply:
yes it might be overkill if it's not expected to scale, let's keep it that way then